src/api/include/migraphx/migraphx.hpp

```diff
@@ -482,38 +482,72 @@ struct program : MIGRAPHX_HANDLE_BASE(program)
     friend bool operator!=(const program& px, const program& py) { return !(px == py); }
 };
 
-inline program parse_onnx(const char* filename, migraphx_onnx_options options)
+struct onnx_options : MIGRAPHX_HANDLE_BASE(onnx_options)
 {
-    return program(make<migraphx_program>(&migraphx_parse_onnx, filename, &options), own{});
+    onnx_options() { this->make_handle(&migraphx_onnx_options_create); }
+
+    onnx_options(migraphx_onnx_options* p, own) { this->set_handle(p, own{}); }
+
+    onnx_options(migraphx_onnx_options* p, borrow) { this->set_handle(p, borrow{}); }
```

Comment:
I dont think we need to borrow references for onnx options so I dont think this constructor is needed.

Reply:
removed